### PR TITLE
Dev 735 rails 3.2

### PR DESCRIPTION
**Pull request by michal-jobseeker**

### CODE REVIEW

File: lib/active_support/cache/cascade_store.rb [hunk 1]
```diff
@@ -88,9 +88,10 @@ def read_entry(key, options) # :nodoc:
           entry = @stores.last.send(:read_entry, key, options)
         else
           synchronize do
-            @stores.detect do |store|
+            store_index = @stores.find_index do |store|
               entry = store.send(:read_entry, key, options) rescue nil
             end
+            write_missing_entry(key, entry, options, store_index) if store_index
```
Comment (cvhoang) [inline lib/active_support/cache/cascade_store.rb]: We should not write to lower cache stores if last_store option is true

File: test/caching_test.rb [hunk 2]
```diff
@@ -234,6 +230,14 @@ def test_really_long_keys
     assert_equal({key => "bar"}, @cache.read_multi(key))
     assert @cache.delete(key)
   end
+
+  def test_missed_read_writes_to_higher_level_stores
+    @store2.write('foo', 'bar')
+    assert_equal 'bar', @store2.read('foo')
```
Comment (cvhoang) [inline test/caching_test.rb]: Do we need this line?

File: lib/active_support/cache/cascade_store.rb (hunk 1 above)
Comment (michal-jobseeker) [inline lib/active_support/cache/cascade_store.rb]: This is in the else branch (when not last_store: true)

Comment (cvhoang) [inline lib/active_support/cache/cascade_store.rb]: Cool. Didn't look at the whole file.

**Pull request merged**